### PR TITLE
Improvements to script for running isolated tests

### PR DESCRIPTION
**Pull request by U-DON**
## Description
- Corrected script to diff starting from the same root as master instead of diffing for all changes. It used to include changes in master were are not made in the current branch, causing more tests to run than necessary.
- More log output for debugging purposes.

## Testing done
Local testing.

## Acceptance criteria
- [ ] The test run should be isolated to apps that were changed within the current branch.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: script/run-isolated-tests.sh [hunk 1]
```diff
@@ -5,10 +5,17 @@
 APP_SUBPATHS="$(find src -name manifest.json | sed -E 's/\/manifest.json//g' | xargs | sed 's/ /|/g')"
 
 # Match changed files to the app paths determined above
-FILES_CHANGED="$(git diff --name-only master)"
-APP_SUBPATHS_CHANGED="$(echo $FILES_CHANGED | grep -oE "($APP_SUBPATHS)")"
+FILES_CHANGED="$(git diff --name-only HEAD $(git merge-base HEAD origin/master))"
```
Comment (bkjohnson) [inline script/run-isolated-tests.sh]: So this diffs the tracked changes on the current branch against the commit where the current branch branched off of master.  I think that's good.

This syntax does the same thing but may be a bit clearer:

```
git diff master...HEAD --name-only
```

File: script/run-isolated-tests.sh [hunk 2]
```diff
@@ -5,10 +5,17 @@
...
+APP_SUBPATHS_CHANGED="$(echo $FILES_CHANGED | grep -oE "($APP_SUBPATHS)" | uniq)"
 NUM_APPS_CHANGED=$(echo "$APP_SUBPATHS_CHANGED" | wc -l)
 
+echo 'Modified files:'
+echo "$FILES_CHANGED"
+echo
+echo 'Modified app dirs:'
+echo "$APP_SUBPATHS_CHANGED"
```
Comment (bkjohnson) [inline script/run-isolated-tests.sh]: For PRs like this one, this variable will be empty.  I'm not a bash expert, but could we do something like this?

```suggestion
[[ ! -z "$APP_SUBPATHS_CHANGED" ]] && echo "$APP_SUBPATHS_CHANGED" || echo "No apps changed"
```

**Review by bkjohnson — APPROVED** [review]
Looks good.  Just a few comments/questions about making things clearer.

File: script/run-isolated-tests.sh [hunk 3]
```diff
@@ -5,10 +5,17 @@
...
+echo
+
 # Run tests only within the apps that have changed.
 # If master is ahead of this branch, tests will still run
 # for apps that have diffs with this branch.
```
[the remark this answers is not in the document]
Comment (U-DON) [reply script/run-isolated-tests.sh]: Yes, it just occurred to me later that I would have to remove this comment.

File: script/run-isolated-tests.sh (hunk 1 above)
Comment (U-DON) [reply script/run-isolated-tests.sh]: Oh, neat!

File: script/run-isolated-tests.sh (hunk 2 above)
Comment (U-DON) [reply script/run-isolated-tests.sh]: There is some more being done in the "no app changes" conditional, so I will move this log below that block, which will exit once its done.

**Review by bkjohnson — COMMENTED** [review]
Do you know why we have 3 `test-and-build` workflows that run for this one commit?
image


As a side-note, I like that Circle shows us this now that the test metadata capture is working:

image

**Comment by bkjohnson** [comment]
[`No changes detected in apps`](https://app.circleci.com/jobs/github/department-of-veterans-affairs/vets-website/1343/parallel-runs/0/steps/0-106) is interesting...

**Comment by U-DON** [comment]
I re-ran the jobs to see it working in real time, so that's why there are multiple runs.

I think I found why the "no changes detected" was happening... It perplexed me at first too. I think I need to use `origin/master` for the diff instead of just `master`.

**Pull request merged**